OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -53,12 +54,20 @@ public WithAttackAnimation(ActorInitializer init, WithAttackAnimationInfo info)
 			armament = init.Self.TraitsImplementing<Armament>()
 				.Single(a => a.Info.Name == info.Armament);
 			wsb = init.Self.Trait<WithSpriteBody>();
+
+			if (init.Self.TraitOrDefault<AttackTurreted>() != null)
```

Comment:
IMO the best option would be to implement this as a separate `WithTurretAttackAnimation` trait, with a `public readonly string Turret = "primary";` yaml key to select the turret that it works on.